types/redux-first-router/index.d.ts

```diff
@@ -152,9 +183,15 @@ export interface Options {
     onAfterChange?<S>(dispatch: Dispatch<S>, getState: StateGetter): void;
     onBackNext?<S>(dispatch: Dispatch<S>, getState: StateGetter): void;
     restoreScroll?(history: History): ScrollBehavior;
+    initialDispatch?: boolean;
+    querySerializer?: QuerySerializer;
     navigators?: NavigatorsConfig;
 }
 
+// Params and Payload differ from Flow definition, as they can be arbitrarily shaped
+export type Params = any;
+export type Payload = any;
+
```

Comment:
I don't think there is a way to express the type for the Payload other than object since the keys of payload depend on string you passed in. I was looking at this https://github.com/faceyspacey/redux-first-router#routesmap and saw that keys in payload are the entries in the input that are preceded with ```:``` I think you want to keep those types as object